pkg/pushsync/pushsync.go

```diff
@@ -352,60 +349,68 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, retryAllo
 			}
 			return nil, fmt.Errorf("closest peer: %w", err)
 		}
-
-		skipPeers = append(skipPeers, peer)
-
-		if ps.skipList.ShouldSkip(peer) {
-			ps.metrics.TotalSkippedPeers.Inc()
-			continue
-		}
-
 		ps.metrics.TotalSendAttempts.Inc()
 
-		go func(peer swarm.Address, ch swarm.Chunk) {
-			ctxd, canceld := context.WithTimeout(ctx, defaultTTL)
-			defer canceld()
+		ctxd, canceld := context.WithTimeout(ctx, defaultTTL)
+		defer canceld()
+		ps.logger.Debugf("pushsync: pushing chunk %s to peer %s", ch.Address().String(), peer.String())
 
-			r, attempted, err := ps.pushPeer(ctxd, peer, ch, retryAllowed)
-			// attempted is true if we get past accounting and actually attempt
-			// to send the request to the peer. If we dont get past accounting, we
-			// should not count the retry and try with a different peer again
-			if attempted {
-				allowedRetries--
-			}
-			if err != nil {
-				logger.Debugf("could not push to peer %s: %v", peer, err)
+		r, attempted, err := ps.pushPeer(ctxd, peer, ch, retryAllowed)
 
-				// if the node has warmed up AND no other closer peer has been tried
-				if ps.warmedUp() && !ps.skipList.HasChunk(ch.Address()) {
-					ps.skipList.Add(peer, ch.Address(), skipPeerExpiration)
+		// attempted is true if we get past accounting and actually attempt
+		// to send the request to the peer. If we dont get past accounting, we
+		// should not count the retry and try with a different peer again
+		if attempted {
+			allowedRetries--
+		}
+		if err != nil {
+			ps.logger.Debugf("pushsync: chunk %s peer %s err: %v, attempted %t, allowedRetries: %d", ch.Address().String(), peer.String(), err, attempted, allowedRetries)
+			var timeToSkip time.Duration
+			switch {
+			case errors.Is(err, context.DeadlineExceeded):
+				// can happen both in NN but also on forwarder nodes.
+				// if its inside the neighborhood - wait a long time.
+				// the originator retry will eventually come in and
+				// would hopefully resolve the situation with the next
+				// closest peer.
+				ps.logger.Debugf("pushsync: chunk %s peer %s context deadline", ch.Address().String(), peer.String())
+				if ps.topologyDriver.IsWithinDepth(ch.Address()) {
+					ps.logger.Debugf("pushsync: chunk %s peer %s context deadline blocking peer for %v", ch.Address().String(), peer.String(), sanctionWait)
+					timeToSkip = sanctionWait
 				}
-
-				select {
-				case resultC <- &pushResult{err: err, attempted: attempted}:
-				case <-ctx.Done():
+			case errors.Is(err, accounting.ErrOverdraft):
+				ps.logger.Debugf("pushsync: chunk %s peer %s err overdraft blocking peer for %v", ch.Address().String(), peer.String(), overdraftWait)
+				timeToSkip = overdraftWait
+			case errors.Is(err, mux.ErrReset):
```

Comment:
mux.ErrReset is when the connection was dropped right?

Would it be better to just santion all peers here?

Reply:
`mux.ErrReset` happens when the downstream peer resets the stream but the connection persists. This is usually because there was some error on the downstream peer or in some other peer further down the cascade (these resets are generally propagated back up the chain)